libs/feature/editor/src/lib/+state/editor.effects.ts

```diff
@@ -0,0 +1,41 @@
+import { inject, Injectable } from '@angular/core'
+import { Actions, createEffect, ofType } from '@ngrx/effects'
+import { of, withLatestFrom } from 'rxjs'
+import { catchError, map, switchMap } from 'rxjs/operators'
+import * as EditorActions from './editor.actions'
+import { EditorService } from '../services/editor.service'
+import { Store } from '@ngrx/store'
+import { selectRecord } from './editor.selectors'
+
+@Injectable()
+export class EditorEffects {
+  private actions$ = inject(Actions)
+  private editorService = inject(EditorService)
+  private store = inject(Store)
+
+  saveRecord$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(EditorActions.saveRecord),
+      withLatestFrom(this.store.select(selectRecord)),
+      switchMap(([, record]) =>
+        this.editorService.saveRecord(record).pipe(
+          map(() => EditorActions.saveRecordSuccess()),
+          catchError((error) =>
+            of(
+              EditorActions.saveRecordFailure({
+                error: error.message,
+              })
+            )
+          )
+        )
+      )
+    )
+  )
+
+  markAsChanged$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(EditorActions.updateRecordField),
+      map(() => EditorActions.markRecordAsChanged())
+    )
+  )
```

Comment:
So my approach here was to avoid putting what could be considered "business logic" inside the reducer. I have the impression that it's good practice to have a very simple reducer, and avoid handling side-effects like that there. This is why I went with an effect, although I agree that it's debatable.

I think I like the fact that an effect is more explicit, then there's no "magic" happening in the reducer